src/releases/ReleasesClient.ts

```diff
@@ -0,0 +1,478 @@
+import {lastValueFrom, map, Observable} from 'rxjs'
+
+import {_action, _getDocument, _getReleaseDocuments} from '../data/dataMethods'
+import type {ObservableSanityClient, SanityClient} from '../SanityClient'
+import type {
+  ArchiveReleaseAction,
+  BaseActionOptions,
+  BaseMutationOptions,
+  CreateReleaseAction,
+  DeleteReleaseAction,
+  EditReleaseAction,
+  HttpRequest,
+  PatchOperations,
+  PublishReleaseAction,
+  RawQueryResponse,
+  ReleaseDocument,
+  SanityDocument,
+  ScheduleReleaseAction,
+  SingleActionResult,
+  UnarchiveReleaseAction,
+  UnscheduleReleaseAction,
+} from '../types'
+import {generateReleaseId} from '../util/createVersionId'
+
+/** @internal */
+export class ObservableReleasesClient {
+  #client: ObservableSanityClient
+  #httpRequest: HttpRequest
+  constructor(client: ObservableSanityClient, httpRequest: HttpRequest) {
+    this.#client = client
+    this.#httpRequest = httpRequest
+  }
+
+  get(
+    {releaseId}: {releaseId: string},
+    options?: {signal?: AbortSignal; tag?: string},
+  ): Observable<ReleaseDocument | undefined> {
+    return _getDocument<ReleaseDocument>(
+      this.#client,
+      this.#httpRequest,
+      `_.releases.${releaseId}`,
+      options,
+    )
+  }
+
+  /**
+   * Creates a new release under the given id, with metadata.
+   *
+   * @param releaseId - The id of the release to create (with no prefix)
+   * @param metadata - The metadata to associate with the release {@link ReleaseDocument}
+   * Must include a `releaseType` {@link ReleaseType}
+   */
+  create(
+    options: BaseActionOptions,
+  ): Observable<SingleActionResult & {releaseId: string; metadata: ReleaseDocument['metadata']}>
+  create(
+    release: {releaseId?: string; metadata?: Partial<ReleaseDocument['metadata']>},
+    options?: BaseActionOptions,
+  ): Observable<SingleActionResult & {releaseId: string; metadata: ReleaseDocument['metadata']}>
+  create(
+    releaseOrOptions?:
+      | {releaseId?: string; metadata?: Partial<ReleaseDocument['metadata']>}
+      | BaseActionOptions,
+    maybeOptions?: BaseActionOptions,
+  ): Observable<SingleActionResult & {releaseId: string; metadata: ReleaseDocument['metadata']}> {
+    const getArgs = (): [
+      string | undefined,
+      Partial<ReleaseDocument['metadata']>,
+      BaseActionOptions | undefined,
+    ] => {
+      if (
+        typeof releaseOrOptions === 'object' &&
+        releaseOrOptions !== null &&
+        ('releaseId' in releaseOrOptions || 'metadata' in releaseOrOptions)
+      ) {
+        return [releaseOrOptions.releaseId, releaseOrOptions.metadata || {}, maybeOptions]
+      }
+
+      return [undefined, {}, releaseOrOptions as BaseActionOptions]
```

Comment:
```suggestion
      return [ generateReleaseId(), {}, releaseOrOptions as BaseActionOptions]
```

And then below you don't need a `finalReleaseId`

Reply:
Thank you!

- [ ] Refactor `getArgs` to look tidier